atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformerJS.kt

```diff
@@ -66,8 +67,10 @@ class AtomicFUTransformerJS(
     private fun AstNode.eraseGetValue(): String {
         val res = this.toSource()
         val primitiveGetValue = MANGLE_VALUE_REGEX
-        val arrayGetValue = ARRAY_GET_VALUE_REGEX
-        return res.replace(arrayGetValue) { matchResult -> "[${matchResult.groupValues[1]}]" }
+        val arrayGetElement = ARRAY_GET_ELEMENT_REGEX
+        val arrayGetElementValue = ARRAY_GET_ELEMENT_VALUE_REGEX
+        return res.replace(arrayGetElementValue) { matchResult -> "[${matchResult.groupValues[1]}]" }
+                  .replace(arrayGetElement) { matchResult -> "[${matchResult.groupValues[1]}]" }
```

Comment:
This does not look right. It is bound to fail in cases like `array[i + j].value`, e.g. when there is complex expression in place of an index. Please, add the corresponding tests, too.

Reply:
Will defer to a later time: https://github.com/Kotlin/kotlinx.atomicfu/issues/71